python/lsst/pipe/tasks/isolatedStarAssociation.py

```diff
@@ -0,0 +1,595 @@
+        dtype=str,
+        default='ra',
+    )
+    dec_column = pexConfig.Field(
+        doc='Name of column with declination.',
+        dtype=str,
+        default='decl',
+    )
+    physical_filter_column = pexConfig.Field(
+        doc='Name of column with physical filter name',
+        dtype=str,
+        default='physical_filter',
+    )
+    band_column = pexConfig.Field(
+        doc='Name of column with band name',
+        dtype=str,
+        default='band',
+    )
+    extra_columns = pexConfig.ListField(
+        doc='Extra names of columns to read and persist (beyond instFlux and error).',
+        dtype=str,
+        default=['x',
+                 'y',
+                 'apFlux_17_0_instFlux',
+                 'apFlux_17_0_instFluxErr',
+                 'apFlux_17_0_flag',
+                 'localBackground_instFlux',
+                 'localBackground_flag']
+    )
+    source_selector = sourceSelectorRegistry.makeField(
+        doc='How to select sources',
+        default='science'
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        source_selector = self.source_selector['science']
+        source_selector.setDefaults()
+
+        source_selector.doFlags = True
+        source_selector.doUnresolved = True
+        source_selector.doSignalToNoise = True
+        source_selector.doIsolated = True
+
+        source_selector.signalToNoise.minimum = 10.0
+        source_selector.signalToNoise.maximum = 1000.0
+
+        flux_flag_name = self.inst_flux_field[0: -len('instFlux')] + 'flag'
+
+        source_selector.flags.bad = ['pixelFlags_edge',
+                                     'pixelFlags_interpolatedCenter',
+                                     'pixelFlags_saturatedCenter',
+                                     'pixelFlags_crCenter',
+                                     'pixelFlags_bad',
+                                     'pixelFlags_interpolated',
+                                     'pixelFlags_saturated',
+                                     'centroid_flag',
+                                     flux_flag_name]
+
+        source_selector.signalToNoise.fluxField = self.inst_flux_field
+        source_selector.signalToNoise.errField = self.inst_flux_field + 'Err'
+
+        source_selector.isolated.parentName = 'parentSourceId'
+        source_selector.isolated.nChildName = 'deblend_nChild'
+
+        source_selector.unresolved.maximum = 0.5
+        source_selector.unresolved.name = 'extendedness'
+
+
+class IsolatedStarAssociationTask(pipeBase.PipelineTask):
+    """Match isolated stars and suchlike.
+    """
+    ConfigClass = IsolatedStarAssociationConfig
+    _DefaultName = 'isolatedStarAssociation'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.makeSubtask('source_selector')
+        # Only log warning and fatal errors from the source_selector
+        self.source_selector.log.setLevel(self.source_selector.log.WARN)
```

Comment:
Probably, but I don't know how (if it is possible to override by default).  This is copied over from how I did it in fgcmcal because otherwise we were inundated with useless warnings from signal-to-noise (I guess really I should file a ticket on the source selector...).